src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,198 @@
+                    while (currentNode.next != null) {
+                        currentNode = currentNode.next;
+                        if ((currentNode.key == null) && (key == null)) {
+                            currentNode.value = value;
+                            size--;
+                            break;
+                        } else {
+                            if ((currentNode.hash == hash)
+                                    && ((currentNode.key == key)
+                                    || (currentNode.key != null && currentNode.key.equals(key)))) {
+                                currentNode.value = value;
+                                size--;
+                                break;
+                            }
+                        }
+                    }
+                    if (currentNode.next == null) {
+                        currentNode.next = new Node<>(hash, key, value, null);
+                    }
+                } else { //esli currentNode.next == 0
+                    currentNode.next = new Node<>(hash, key, value, null);
+                }
+            }
+        }
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> element;
+        if (size == 0) {
+            return null;
+        }
+        for (int i = 0; i < table.length; i++) {
+            element = table[i];
+            if (element == null) {
+                continue;
+            }
+            if (element.key == null && key == null) {
+                return element.value;
+            } else if ((element.hash == hash(key))
+                    && ((element.key == key)
+                    || (element.key != null && element.key.equals(key)))) {
+                return element.value;
+            }
+            while (element.next != null) {
+                element = element.next;
+                if (element == null) {
+                    continue;
+                }
+                if (element.key == null && key == null) {
+                    return element.value;
+                } else if ((element.hash == hash(key))
+                        && ((element.key == key)
+                        || (element.key != null && element.key.equals(key)))) {
+                    return element.value;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        int h;
+        return (key == null) ? 0 : Math.abs((h = key.hashCode()) ^ (h >>> 16));
+    }
+
+    private Node<K, V>[] resize() {
+        Node<K, V>[] oldTab = table;
+        Node<K, V>[] newTab;
+        int oldCapacity;
+        int newCapacity;
+        int oldThreshold = threshold;
+        int newThreshold;
+        oldCapacity = (oldTab == null) ? 0 : oldTab.length;
```

Comment:
```suggestion
        oldCapacity = oldTab.length;
```
oldTab can't be null if you create table it in a constructor

Reply:
i should change my logic to do this(